### PR TITLE
Feature: Support custom data dir

### PR DESCRIPTION
**Pull request by timotheeguerin**

### CODE REVIEW

File: src/services/git-fetch/git-fetch.service.ts [hunk 1]
```diff
@@ -18,19 +19,20 @@ export const defaultFetchOptions: FetchOptions = {
...
+
   private currentFetches = new Map<string, Promise<Repository>>();
   private lastFetch = new Map<string, number>();
 
   private cacheReady: Promise<string>;
 
-  constructor(fs: FSService) {
-    this.cacheReady = fs.mkdir(repoCacheFolder);
+  constructor(fs: FSService, private config: Configuration) {
+    this.cacheReady = fs.mkdir(this.repoCacheFolder);
```
Comment (vladbarosan) [inline src/services/git-fetch/git-fetch.service.ts]: is this mkdir called only on startup ?
preferably no sync calls for IO

File: src/services/repo/repo.service.ts [hunk 2]
```diff
@@ -142,4 +135,11 @@ export class RepoService {
       }),
     );
   }
+
+  private getRepoMainPath(remote: string, namespace?: string) {
```
Comment (vladbarosan) [inline src/services/repo/repo.service.ts]: what is the purpose of namespace

File: src/services/git-fetch/git-fetch.service.ts (hunk 1 above)
Comment (timotheeguerin) [reply src/services/git-fetch/git-fetch.service.ts]: yes and its async

File: src/services/repo/repo.service.ts [hunk 3]
```diff
@@ -142,4 +135,11 @@ export class RepoService {
...
+    if (namespace) {
+      return path.join(this.fetchService.repoCacheFolder, namespace, encodeURIComponent(remote));
```
Comment (vladbarosan) [inline src/services/repo/repo.service.ts]: will this long path cause any issue on win ?

File: src/services/repo/repo.service.ts (hunk 2 above)
Comment (timotheeguerin) [reply src/services/repo/repo.service.ts]: single repos vs compare repos, just for group

File: src/services/repo/repo.service.ts (hunk 3 above)
Comment (timotheeguerin) [reply src/services/repo/repo.service.ts]: it cause the issue either way, its broken on windows depending on the repo

**Pull request merged**